### PR TITLE
Adding solution of strassen algorithm.

### LINKED ISSUE #3624: [Feature Request]: Implementing Strassen's Algorithm in C++ under DSA folder.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I propose adding the implementation of Strassen's Algorithm in C++ to the DSA (Data Structures and Algorithms) folder. Strassen's Algorithm is an efficient matrix multiplication algorithm that reduces the time complexity compared to the conventional matrix multiplication approach.

### Use Case

This feature would enhance the project by providing an efficient algorithm for matrix multiplication, which is a common requirement in various computational tasks. For example, in scenarios involving large-scale data processing or scientific computations, Strassen's Algorithm can significantly improve performance.

### Benefits

Implementing Strassen's Algorithm in C++ under the DSA folder will provide a faster matrix multiplication method, significantly reducing computational time for large matrices. This will enhance learning opportunities for developers and students by showcasing advanced matrix multiplication techniques.

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by sjain1909**
## Fixing Issue 

Fix #3624

## Description

This pull request adds the implementation of Strassen's algorithm in C++ to the DSA folder. Strassen's algorithm is an efficient algorithm for matrix multiplication that is faster than the conventional matrix multiplication algorithm. It reduces the time complexity from O(n^3) to approximately O(n^2.81), making it suitable for large matrices.

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

Strassen's algorithm implementation includes:
- Efficient matrix multiplication using a divide-and-conquer approach.
- Reduction in time complexity compared to the conventional algorithm.
- Clear and well-commented code for easy understanding and maintenance.

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Markdown Guide](https://www.markdownguide.org/)

### CODE REVIEW

**Comment by sjain1909** [comment]
@Ajay-Dhangar sir, kindly review and let me know if any changes required.

File: dsa/Advance/strassen-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    return 0;
+}
+    ```
+  </TabItem>  
+</Tabs>
+
+### Complexity Analysis
+
+- **Time Complexity:** \(O(n^{\log_2 7}) \approx O(n^{2.81})\)
```
Comment (Ajay-Dhangar) [inline dsa/Advance/strassen-algorithm.md:165]: replace`\(O(n^{\log_2 7}) \approx O(n^{2.81})\)` with `$O(n^{\log_2 7}) \approx O(n^{2.81})$`

File: dsa/Advance/strassen-algorithm.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+- **Space Complexity:** \(O(n^2)\)
```
Comment (Ajay-Dhangar) [inline dsa/Advance/strassen-algorithm.md:166]: replace `\(O(n^2)\)` to `$O(n^2)$`

File: dsa/Advance/strassen-algorithm.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+The time complexity is significantly reduced compared to the traditional \(O(n^3)\) matrix multiplication. The space complexity is determined by the storage requirements of the matrices.
```
Comment (Ajay-Dhangar) [inline dsa/Advance/strassen-algorithm.md:168]: replace `\(O(n^3)\)` with `$O(n^3)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by sjain1909** [comment]
@Ajay-Dhangar please close this request as done via #3671.

**Pull request closed without merging**